core/appeal/service.go

```diff
@@ -1003,18 +1004,26 @@ func (s *Service) CreateAccess(ctx context.Context, a *domain.Appeal, opts ...Cr
 	return nil
 }
 
-func (s *Service) checkExtensionEligibility(a *domain.Appeal, p *domain.Provider, activeGrants map[string]map[string]map[string]*domain.Grant) error {
+func (s *Service) checkExtensionEligibility(a *domain.Appeal, p *domain.Provider, policy *domain.Policy, activeGrants map[string]map[string]map[string]*domain.Grant) error {
 	if activeGrants[a.AccountID] != nil &&
 		activeGrants[a.AccountID][a.ResourceID] != nil &&
 		activeGrants[a.AccountID][a.ResourceID][a.Role] != nil {
 		if p.Config.Appeal != nil {
-			if p.Config.Appeal.AllowActiveAccessExtensionIn == "" {
+			// Default to use provider config if policy config is not set
+			AllowActiveAccessExtensionIn := p.Config.Appeal.AllowActiveAccessExtensionIn
+			if policy != nil &&
+				policy.AppealConfig != nil &&
+				policy.AppealConfig.AllowActiveAccessExtensionIn != "" {
+				AllowActiveAccessExtensionIn = policy.AppealConfig.AllowActiveAccessExtensionIn
+			}
+
+			if AllowActiveAccessExtensionIn == "" {
 				return ErrAppealFoundActiveGrant
 			}
```

Comment:
I think we could also return error early if `AllowActiveAccessExtensionIn` in empty
```suggestion
			// Default to use provider config if policy config is not set
			AllowActiveAccessExtensionIn := p.Config.Appeal.AllowActiveAccessExtensionIn
			if AllowActiveAccessExtensionIn == "" {
				return ErrAppealFoundActiveGrant
			}
			
			if policy != nil &&
				policy.AppealConfig != nil &&
				policy.AppealConfig.AllowActiveAccessExtensionIn != "" {
				AllowActiveAccessExtensionIn = policy.AppealConfig.AllowActiveAccessExtensionIn
			}

```